### ISSUE TITLE
Energy Conduits not transferring power when replaced

### ISSUE THREAD

**Issue by Xefyr0** [AlmostReliable/kubejs-enderio#9]
### Minecraft Version

1.20.1

### Mod Version

Modpack: Monifactory 0.10.4; KubeJS EnderIO 0.6.0

### Describe the bug!

After replacing a base full of KJS energy conduits, there were difficulties getting energy to transfer.

First, a relog was necessary to get the AE2 system powered:
https://www.twitch.tv/videos/2314611340?t=5h12m31s

Second, many other conduit interfaces had to have "Insert" toggled:
https://www.twitch.tv/videos/2314611340?t=5h21m1s

Disclaimer: I'm not the person depicted in the VOD. I'm one of the modpack's developers instead.
I'm happy to answer any questions about the issue, though!

### Additions and Modifications

No

### Did the issue happen in singleplayer or on a server?

Server

**Comment by rlnt** [comment]
The upgrade replacement logic is handled by EnderIO itself. It's not done by the add-on.
I could probably debug this issue and PR a fix to EnderIO when I find the time though.

Were you able to reproduce this in a smaller example?

**Comment by Xefyr0** [comment]
Yes actually - it was surprisingly simple. I had the impression that this bug wouldn't be very consistent for some reason.
![reproducible](https://github.com/user-attachments/assets/3914a177-4db3-4524-b158-d53cf6702329)

In the lastest version of Monifactory, I placed down an AE2 controller, Energetic Alloy Energy Conduit, and Numismatic Dynamo as shown in the image.
Next, I right-clicked on the conduit with a Vibrant Alloy Energy conduit, replacing it.
Jade then showed the AE2 controller using up its internal energy buffer without being replenished by the conduit.

**Comment by Xefyr0** [comment]
I should have done this earlier, but this bug is also reproducible with just KubeJS EnderIO and its dependencies.
Thus, we can rule out interactions with other mods as a potential cause. Both the above recreation and this recreation use the same script:

```
/**
 * Define custom EnderIO conduit tiers and their flow rate
 * using KubeJS-EnderIO.
 */
EnderIOEvents.conduits(event => {
    event.registerEnergyConduit("conductive_conduit", "Conductive Iron Energy Conduit", 512)
    event.registerEnergyConduit("energetic_conduit", "Energetic Alloy Energy Conduit", 2048)
    event.registerEnergyConduit("vibrant_conduit", "Vibrant Alloy Energy Conduit", 8192)
    event.registerEnergyConduit("endsteel_conduit", "End Steel Energy Conduit", 32768)
    event.registerEnergyConduit("lumium_conduit", "Lumium Energy Conduit", 131072)
    event.registerEnergyConduit("signalum_conduit", "Signalum Energy Conduit", 524288)
    event.registerEnergyConduit("enderium_conduit", "Enderium Energy Conduit", 2097152)
    event.registerEnergyConduit("cryolobus_conduit", "Cryolobus Energy Conduit", 8388608)
    event.registerEnergyConduit("sculk_superconductor_conduit", "Sculk Superconductor Energy Conduit", 134217728)
})
```

![test](https://github.com/user-attachments/assets/461f0c17-721f-4014-b744-5c3f9cb7c1dc)

**Comment by rlnt** [comment]
That's a good minimal reproduction setup, thank you. This is good to know. I already talked to Rover, and have some ideas, but I can't tell you an ETA. I don't have a lot of time at the moment.

**Comment by rlnt** [comment]
After revisiting this, it seems to be a bug in the replacement logic of EnderIO. You can easily reproduce that with fluid conduits as well.

Nothing I can do about this. The logic for that is *very* deeply nested inside EnderIO.
